app/controllers/report_controller/widgets.rb

```diff
@@ -655,7 +655,17 @@ def widget_set_record_vars(widget)
     #      end
     #      widget.miq_widget_shortcuts = ws
     #    end
-    widget.options[:url] = @edit[:new][:url] unless @edit[:new][:url].blank?
+    if @sb[:wtype] == "rf"
```

Comment:
The wtype probably stands for widget type. RF may stand for RSSFeed.

Look, I agree that there is a room for improvements in this controller. My point was just to fix a bug. If you want me to commit to greater refactoring here. I'll see what I can do.